server/controllers/finance/reports/cashflow/index.js

```diff
@@ -461,42 +553,263 @@ function report(req, res, next) {
+
+          localCashGlobals = incomesGlobals;
+        }
+
+        let operatingGlobalsTextKeys;
+        let operatingGlobalsTotalByTextKeys;
+        let operatingGlobals;
+        let totalOperatingExpense;
+        let sumOperatingExpense = 0;
+
+        if (referenceAccountsOperating.length) {
+          /**
+           * Getting the total expenses that correspond to local expenses for each period
+           */
+          const opReferenceGroups = operatingReferenceAccounts.map(item => item.referenceGroup);
+          /**
+           * Getting the key for operating expenses
+           */
+          operatingGlobalsTextKeys = opReferenceGroups;
+
+          /** Filtering the items that make up operating expenses from the total expenses */
+          operatingGlobalsTotalByTextKeys = Object.fromEntries(
+            opReferenceGroups.map(
+              key => [key, expenseGlobalsTotalByTextKeys[key]]).filter(([value]) => value !== undefined,
+            ),
+          );
+
+          // Data filtering
+          operatingGlobalsTotalByTextKeys = Object.fromEntries(
+            Object.entries(operatingGlobalsTotalByTextKeys).filter((entry) => {
+              return entry[1] !== undefined;
+            }),
+          );
+
+          operatingGlobalsTextKeys = operatingGlobalsTextKeys.filter(
+            item => Object.keys(operatingGlobalsTotalByTextKeys).includes(item));
+
+          /**
+           * This is the calculation of the sum of operating expenses for each period
+           */
+          totalOperatingExpense = cashflowFunction.aggregateData(operatingGlobalsTotalByTextKeys);
+
+          /**
+           * This is the sum of local revenues
+           */
+          sumOperatingExpense = totalOperatingExpense.sumAggregate;
+
+          operatingGlobals = expensesGlobals;
+        }
+
+        let personnelGlobalsTextKeys;
+        let personnelGlobalsTotalByTextKeys;
+        let personnelGlobals;
+        let totalPersonnelExpense;
+        let sumPersonnelExpense = 0;
+
+        if (referenceAccountsPersonnel.length) {
+          /**
+           * Getting the total expenses related to Personnel expense for each period
+           */
+
+          const personnelReferenceGroups = personnelReferenceAccounts.map(item => item.referenceGroup);
+          personnelGlobalsTextKeys = personnelReferenceGroups;
+
+          /**
+           * Extraction of references related to personnel expenses and data filtering
+           */
+          personnelGlobalsTotalByTextKeys = Object.fromEntries(
+            personnelReferenceGroups.map(
+              key => [key, expenseGlobalsTotalByTextKeys[key]]).filter(([value]) => value !== undefined,
+            ),
+          );
+
+          personnelGlobalsTotalByTextKeys = Object.fromEntries(
+            Object.entries(personnelGlobalsTotalByTextKeys).filter((entry) => {
+              return entry[1] !== undefined;
+            }),
+          );
```

Comment:
This also uses the mysterious `return entry[1] !== undefined;` signature.

Reply:
We apply this filter to avoid having empty or invalid entries in the final results, and to ensure that the cashflow report remains reliable and clean.

Reply:
The global and synthetic cashflow reports are designed to have sections, and within these sections, we place account references.
It can happen that there are no transactions for a given account reference.
The filter on entry[1] checks whether there is any data associated with the key.
If entry[1] is undefined, the corresponding key entry[0] will not appear in the final report.
This ensures that the report only displays sections with actual data, keeping it clean, accurate, and easy to read.